kura/org.eclipse.kura.wire.helper.provider/src/main/java/org/eclipse/kura/internal/wire/helper/WireHelperServiceImpl.java

```diff
@@ -144,6 +144,11 @@ public WireSupport newWireSupport(final WireComponent wireComponent,
             return null;
         }
 
+        if (wireComponentRef.getProperty(SERVICE_PID) instanceof java.util.ArrayList) {
```

Comment:
Maybe we can change this to

` (!(wireComponentRef.getProperty(SERVICE_PID) instanceof String))`

since the code expects the service pid to be a string, in this way we can probably catch more error cases.

Reply:
I had a look at it, thanks.
Imo if we catch the specific case ArrayList, we can give a strong hint message to the developer ("Maybe you used the same PID twice in OSGi metadata?")

The pid is always string, because it is parsed from the META-INF file. It just happens to be an ArrayList, if it is defined twice by error. You kind of loose this know-how here if you go generic.

Should I make it generic or remain specific? :)